soc: espressif: esp32s3: Adjust DRAM size for .bss.sector_buffers to prevent overflow with larger images

FacepalmMute opened the pull request:
This pull request solves the issue descried in https://github.com/zephyrproject-rtos/zephyr/issues/80138

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/80138

marekmatej commented on soc/espressif/esp32s3/memory.h:
```diff
@@ -47,9 +48,22 @@
  */
 #define BOOTLOADER_USER_DRAM_END DRAM_BUFFERS_START
 
+/* Calculate required DRAM size for .bss.sector_buffers to adjust BOOTLOADER_DRAM_SEG_LEN
```
I think it is much more straightforward to simply increase the `BOOTLOADER_DRAM_SEG_LEN` value to support the "worst case" scenario.
All the MCUboot segments - except the `BOOTLOADER_IRAM_LOADER` - will be rewritten by the last stage of the bootloader anyway.

FacepalmMute replied:
Simple is good 👍

In the worst-case scenario, we could have an ESP32S3 N32R8V with 32MB of flash. A crazy user might allocate approximately 16MB for the image partition. This would mean that sector_buffers might use around 0x10000 in the worst case.

So can i just add `0x10000` to `BOOTLOADER_DRAM_SEG_LEN` ?

marekmatej replied:
Since we can't support random "crazy user" let's be realistic now. The biggest partition size for the 32MB flash would be ~12MB, which is an addition of `0xC000` to the actual DRAM size, so with `0x15000` we should be fine.

marekmatej reviewed (changes requested):
Good approach in general, but let's make it simpler.

Commit:
soc: espressif: esp32s3: Adjust BOOTLOADER_DRAM_SEG_LEN for worst case

Larger image partitions require more space in DRAM due to
the increase in .bss.sector_buffers.
Each sector in .bss.sector_buffers consumes 16 bytes.
In the worst case scenario, such as with the ESP32S3 N32R8V,
which has 32 MB of flash and most likely 12 MB image partition,
an addition of 0xc000 should be sufficient to accommodate this.

Signed-off-by: Nik Schewtschuk <nikita.schewtschuk@smartmechatronics.de>
